core/src/operations/decorators/advice.rs

```diff
@@ -1,11 +1,33 @@
 use core::fmt;
 
 use super::SignatureKind;
-use crate::Felt;
 
 // ADVICE INJECTORS
 // ================================================================================================
 
+// Randomly generated constant values for the VM's native events. All values were sampled between 0
+// and 2^32.
+
+pub const EVENT_MERKLE_NODE_MERGE: u32 = 191808474;
+pub const EVENT_MERKLE_NODE_TO_STACK: u32 = 2389745239;
+pub const EVENT_UPDATE_MERKLE_NODE: u32 = 3895296046;
+pub const EVENT_MAP_VALUE_TO_STACK: u32 = 3754237093;
+pub const EVENT_MAP_VALUE_TO_STACK_N: u32 = 2928255101;
+pub const EVENT_U64_DIV: u32 = 4194199493;
+pub const EVENT_EXT2_INV: u32 = 1774099307;
+pub const EVENT_EXT2_INTT: u32 = 1091345727;
+pub const EVENT_SMT_PEEK: u32 = 3186495057;
+pub const EVENT_U32_CLZ: u32 = 1180644394;
+pub const EVENT_U32_CTZ: u32 = 2777457657;
+pub const EVENT_U32_CLO: u32 = 2311971056;
+pub const EVENT_U32_CTO: u32 = 3070738514;
+pub const EVENT_ILOG2: u32 = 3092594227;
+pub const EVENT_MEM_TO_MAP: u32 = 3593064965;
+pub const EVENT_HDWORD_TO_MAP: u32 = 2433155432;
+pub const EVENT_HDWORD_TO_MAP_WITH_DOMAIN: u32 = 233822648;
+pub const EVENT_HPERM_TO_MAP: u32 = 2997735825;
+pub const EVENT_FALCON_SIG_TO_STACK: u32 = 3099530921;
```

Comment:
Not sure these need to be random values. If we make them ordered, would be easy to tell if there duplicates.

Reply:
The idea behind making them random is it makes negligible the probability that a library accidentally reuses one of the native event values. That's also what we'll strongly recommend libraries do, so I thought it'd be good to practice what we preach.